modules/core/CMakeLists.txt

```diff
@@ -167,6 +167,10 @@ if(HAVE_HPX)
   ocv_target_link_libraries(${the_module} LINK_PRIVATE "${HPX_LIBRARIES}")
 endif()
 
+if(HAVE_OPENMP)
+    ocv_target_link_libraries(${the_module} PUBLIC ${OpenMP_CXX_LIBRARIES} OpenMP::OpenMP_CXX)
```

Comment:
Using both approaches looks error prone.

Reply:
You mean doing this here and in `OpenCVFrameworks.cmake`. Might be the latter can be omitted; i was just looking for a place to add the link_libraries call. There wasn't any in the `OpenCVFrameworks.cmake` file and I have no idea about the proper place. Where do you suggest to put this?

Reply:
And I guess here there would be a problem if the library target does not exist in older cmake versions. I assume an `if(TARGET…`  would be sufficient?

Reply:
Yes, this should work:

```
if(TARGET OpenMP::OpenMP_CXX)
  ... use modern approach ...
else()
  ... use legacy approach (keep existed behavior the same) ...
endif()
```

Reply:
fixed